src/lib/rules/apple-touch-icons/apple-touch-icons.ts

```diff
@@ -0,0 +1,268 @@
+
+            if (appleTouchIcons.length === 0) {
+                await context.report(resource, null, `No 'apple-touch-icon' was specified`);
+
+                return;
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Choose the icon that will most likely
+            // pass most of the following tests.
+
+            const appleTouchIcon: IAsyncHTMLElement = chooseBestIcon(appleTouchIcons);
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Check if `rel='apple-touch-icon'`.
+            // See `getAppleTouchIcons` function for more details.
+
+            if (normalizeString(appleTouchIcon.getAttribute('rel')) !== 'apple-touch-icon') {
+                await context.report(resource, appleTouchIcon, `'rel' attribute value should be 'apple-touch-icon'`);
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Since we are recommending just one icon, the `sizes`
+            // attribute is not needed. Also, pre-4.2 versions of iOS
+            // ignore the `sizes` attribute.
+            //
+            // https://mathiasbynens.be/notes/touch-icons
+            // https://html.spec.whatwg.org/multipage/semantics.html#attr-link-sizes
+
+            if (appleTouchIcon.getAttribute('sizes')) {
+                await context.report(resource, appleTouchIcon, `'sizes' attribute is not needed`);
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Check if the `apple-touch-icon` exists, is the right
+            // image format, the right size, etc.
+
+            await checkImage(appleTouchIcon, resource);
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Check if the `apple-touch-icon` is included in the `<body>`.
+
+            const bodyAppleTouchIcons: Array<IAsyncHTMLElement> = getAppleTouchIcons(await pageDOM.querySelectorAll('body link'));
+
+            for (const icon of bodyAppleTouchIcons) {
+                if (icon.isSame(appleTouchIcon)) {
+                    await context.report(resource, appleTouchIcon, `'apple-touch-icon' should be specified in the '<head>'`);
+                }
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // All other `apple-touch-icon`s should not be included.
+
+            for (const icon of appleTouchIcons) {
+                if (!icon.isSame(appleTouchIcon)) {
+                    await context.report(resource, icon, `A 'apple-touch-icon' was already specified`);
+                }
+            }
+        };
+
+        return { 'traverse::end': validate };
+    },
+
+    meta: {
+        docs: {
+            category: `PWAs`,
+            description: `Require an 'apple-touch-icon'`
+        },
+        fixable: 'code',
+        ignoredConnectors: ['chrome'],
```

Comment:
@molant The _"is this a PNG?", "is it 180x180px?"_, etc, will not work with the `chrome` connector as they require the raw bytes of the image. :(

Reply:
Why can't you use [`rawContent`](https://github.com/sonarwhal/sonar/blob/master/src/lib/connectors/shared/remote-debugging-connector.ts#L294)?

Reply:
[I am](https://github.com/sonarwhal/sonar/pull/488/files#diff-046ee208e76bb5dffb66c554f2aac000R123). Now that you mention it, `rawContent` used to work for `chrome`, something must be wrong.

Reply:
The problem with `rawContent` is [here](https://github.com/sonarwhal/sonar/blob/master/src/lib/connectors/shared/remote-debugging-connector.ts#L798). Nothing is connected when using `fetchContent`.

I'll create an issue for that and work on it tomorrow.

Reply:
OK.

Reply:
This is the issue https://github.com/sonarwhal/sonar/issues/495